### PR TITLE
tools: support versioned node shared libraries on z/OS

### PR DESCRIPTION
**Pull request by alexcfyung**
The shared libraries will now be stores in lib.target as opposed to
obj.target, libnode.version.so, libnode.x (for npm backwards compat and
testing), and libnode.version.x (for builds). The install will also
include libnode.so link that points to libnode.version.so (this will be
used by native npms for backwards compat).

This PR will add a new file at `tools/zos/modifysidedeck.sh`.

This PR is depend on https://github.com/ibmruntimes/node-gyp/pull/2 and https://github.com/ibmruntimes/gyp-next/pull/1.

##### Checklist

- [ ] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

### CODE REVIEW

File: tools/install.py [hunk 1]
```diff
@@ -120,6 +121,17 @@ def corepack_files(action):
 #   'pnpx': 'dist/pnpx.js',
   })
 
+  # On z/OS, we install node-gyp for convenience as native add-ons
+  # are dependend on
```
Comment (richardlau) [inline tools/install.py]: Can you expand on why this is needed on z/OS? The regular developer flows for installing modules normally have `npm` automatically run `node-gyp` from its own dependencies either automatically on `npm install` (if no install script was defined and a gyp file found) or by the individual modules scripting a call to it as a devdependency.

**Comment by richardlau** [comment]
Also the lint-sh failures will need to be addressed.

File: tools/install.py (hunk 1 above)
Comment (alexcfyung) [reply tools/install.py]: It is because some vendors don't have external access and may want to build their own native node.js modules

File: tools/install.py [hunk 2]
```diff
@@ -122,7 +122,8 @@ def corepack_files(action):
   })
 
   # On z/OS, we install node-gyp for convenience as native add-ons
-  # are dependend on
+  # are dependend on, as some vendors don't have external access
```
Comment (gabylb) [inline tools/install.py]: typo: dependend -> depended

File: tools/install.py [hunk 3]
```diff
@@ -122,7 +122,8 @@ def corepack_files(action):
...
+  # and may want to build their own native node.js modules
```
Comment (gabylb) [inline tools/install.py]:
```suggestion
  # and may want to build native addon modules.
```
because the reason applies to any native addon, not just their own.

File: tools/zos/modifysidedeck.sh [hunk 4]
```diff
@@ -12,19 +12,22 @@ originalsidedeck=$1
 outputsidedeck=$2
 newdllname=$3
 
+SCRIPT_DIR=$(dirname $0)
 ID=`date +%C%y%m%d_%H%M%S`
-TMP="/tmp/sidedeck-$(basename $0).$ID.tmp"
-TMP2="/tmp/sidedeck-$(basename $0).$ID.tmp.2"
+TMP="/tmp/sidedeck-$(basename "$0").$ID.tmp"
+TMP2="/tmp/sidedeck-$(basename "$0").$ID.tmp.2"
```
Comment (gabylb) [inline tools/zos/modifysidedeck.sh:18]: Double quotes inside double quotes? I think the double quotes are instead needed for the SCRIPT_DIR and ID assignments.

File: tools/zos/modifysidedeck.sh [hunk 5]
```diff
@@ -12,19 +12,22 @@ originalsidedeck=$1
...
 
 # Remove on exit/interrupt
-trap "/bin/rm -rf $TMP $TMP2 && exit" EXIT INT TERM QUIT HUP
+trap '/bin/rm -rf "$TMP" "$TMP2" && exit' EXIT INT TERM QUIT HUP
```
Comment (gabylb) [inline tools/zos/modifysidedeck.sh:21]: The double quote inside the single quote won't be interpreted. Is this change needed?

Comment (alexcfyung) [reply tools/zos/modifysidedeck.sh:21]: yes that was 1 of linter's complain, and I think we do not want it to be interpreted when running this line but instead when trapping one of the signal

File: tools/zos/modifysidedeck.sh (hunk 4 above)
Comment (alexcfyung) [reply tools/zos/modifysidedeck.sh:18]: it was one of linter's complain, and within `$()` the outside quote does not apply

File: tools/install.py [hunk 6]
```diff
@@ -120,6 +121,18 @@ def corepack_files(action):
 #   'pnpx': 'dist/pnpx.js',
   })
 
+  # On z/OS, we install node-gyp for convenience as native add-ons
+  # are depended on, as some vendors don't have external access
+  # and may want to build native addon modules.
```
Comment (gabylb) [inline tools/install.py]:
```suggestion
  # On z/OS, we install node-gyp for convenience, as some vendors don't have
  # external access and may want to build native addons.
```

**Review by gabylb — APPROVED** [review]
LGTM
